src/Bicep.Core/Semantics/Namespaces/SystemNamespaceSymbol.cs

```diff
@@ -504,6 +505,46 @@ private static SyntaxBase StringLiteralFunctionReturnTypeEvaluator(FunctionCallS
             return SyntaxFactory.CreateStringLiteral(stringLiteral.RawStringValue);
         }
 
+        private static TypeSymbol JsonTypeBuilder(IBinder binder, IFileResolver fileResolver, IDiagnosticWriter diagnostics, ImmutableArray<FunctionArgumentSyntax> arguments, ImmutableArray<TypeSymbol> argumentTypes)
+        {
+            static TypeSymbol ToBicepType(JToken token)
+                => token switch {
+                    JObject @object => new ObjectType(
+                        "object",
```

Comment:
The type name "object" isn't super descriptive, but we don't really have anything better to provide - this should improve with #typesystemv2